ts/component/tab-bar-builder.ts

```diff
@@ -0,0 +1,129 @@
+import { Pair } from "../util/pair";
+import { getSearchParams, updateUrlParams } from "../util/urls";
+import { TableBuilder } from "./table-builder";
+
+class Tab {
+  tabContainer: HTMLButtonElement;
+  table: Pair<any, string>;
+  selected: boolean = false;
+  onSelect: (table: Pair<any, string>) => void;
+  constructor(
+    table: Pair<any, string>,
+    onSelect: (table: Pair<any, string>) => void,
+    selected?: boolean
+  ) {
+    this.table = table;
+    this.onSelect = onSelect;
+    const button = document.createElement("button");
+    if (selected) this.selected = selected; // is the default tab
+    button.className =
+      "font-inter font-medium text-12 text-black bg-white px-2 py-1 rounded-md";
+    button.textContent = table.value;
+    button.onclick = () => {
+      console.log(`clicked tab ${table.value}`);
+      if (!this.selected) {
+        this.selected = !this.selected;
+        this.styleButton();
+        onSelect(this.table);
+      }
+    };
+    this.tabContainer = button;
+    this.styleButton();
+  }
+
+  // modifies the tab styling according to whether or not it has been selected
+  public styleButton() {
+    if (this.selected) {
+      this.tabContainer.classList.replace("text-black", "text-green-200");
+      this.tabContainer.classList.add(
+        "ring-2",
+        "ring-green-200",
+        "ring-offset-1"
+      );
+    } else {
+      this.tabContainer.classList.replace("text-green-200", "text-black");
+      this.tabContainer.classList.remove(
+        "ring-2",
+        "ring-green-200",
+        "ring-offset-1"
```

Comment:
`ring-green-200` and `ring-offset-1` are re-added whether selected or not. Can add them when creating the button and then never remove them